src/client/hooks/useSelectedSellToken.ts

```diff
@@ -0,0 +1,46 @@
+import { keyBy } from 'lodash';
+
+import { getConfig } from '@config';
+import { useAppSelector } from '@src/client/hooks';
+import { TokensSelectors } from '@src/core/store';
+import { GeneralLabView, GeneralVaultView, TokenView } from '@src/core/types';
+import { createPlaceholderToken } from '@src/utils';
+
+interface SelectedSellTokenProps {
+  selectedSellTokenAddress?: string;
+  selectedVaultOrLab?: GeneralVaultView | GeneralLabView;
+  allowTokenSelect?: boolean;
+}
+
+interface SelectedSellToken {
+  selectedSellToken?: TokenView;
+  sourceAssetOptions: TokenView[];
+}
+
+export const useSelectedSellToken = ({
+  selectedSellTokenAddress,
+  selectedVaultOrLab,
+  allowTokenSelect,
+}: SelectedSellTokenProps): SelectedSellToken => {
+  const { ETHEREUM_ADDRESS } = getConfig();
+  const selectedTokenMap = useAppSelector(TokensSelectors.selectTokensMap);
+  let userTokens = useAppSelector(TokensSelectors.selectZapInTokens);
+  userTokens = selectedVaultOrLab?.allowZapIn ? userTokens : [];
+
+  if (!selectedSellTokenAddress) return { selectedSellToken: undefined, sourceAssetOptions: [] };
```

Comment:
nit: Unsure if this was added for readability, but in these cases we can just omit the property
```suggestion
  if (!selectedSellTokenAddress) return { sourceAssetOptions: [] };
```
No action needed, just a thought